### ISSUE TITLE
`tailscale serve` doesn't handle emoji in URL path segments

### ISSUE THREAD

**Issue by Xe** [tailscale/tailscale#7027]
### What is the issue?

To be fair, the exact semantics of how you are supposed to handle astral plane UTF-8 characters like emoji are undefined and vary from implementation to implementation. However Chromium, Firefox and Safari seem to percent-encode the URL when sending it and display it as the emoji.

### Steps to reproduce

```
tailscale serve /.within/🥺 path /home/cadey/tmp/xn--ts9h
```

### OS

Linux

### OS version

NixOS Unstable

### Tailscale version

1.34.2

### Bug report

BUG-4fa55a51a4af084dc37f75e704e3328978b10e8db82b78dc3b3ed0eccda27180-20230120194417Z-cdbb4f6be6d40b5c

**Comment by bradfitz** [comment]
The `xn--ts9h` in your bug text is a little misleading, as this bug would have nothing to do with punycode/IDNA.

(Lest anybody get misled reading this later)

**Comment by Xe** [comment]
The `xn--ts9h` is the name of the folder that I put the files in because `scp` didn't like folder paths with emoji in them.

**Comment by bradfitz** [comment]
Yeah, I'm just clarifying for future readers of this bug that that part is irrelevant. It could be /home/cadey/tmp/foo as far as this bug is concerned.
